### PR TITLE
update week0004

### PR DESCRIPTION
**Pull request by truongngodang**
ok?

### CODE REVIEW

**Comment by vietlq** [comment]
This looks better ;)

File: week0004/Week0004.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class Week0004 {
+
+    public static void main(String[] args) {
+        ArrayList<Guest> guestsEvent1 = ReaderCVS.read(args[0]);
```
Comment (vietlq) [inline week0004/Week0004.java]: Check if you have exactly 2 files first

File: week0004/Week0004.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        String line = "";
+        String cvsSplitBy = ",";
+        ArrayList<Guest> guests = new ArrayList<Guest>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(new File(csvFilename)));
+            for (int i = 0; (line = br.readLine()) != null; i++) {
+                if (i > 0) {
+                    // use comma as separator
+                    String[] guest = line.split(cvsSplitBy);
+                    guests.add(new Guest(guest[0], guest[1], guest[2]));
```
Comment (vietlq) [inline week0004/Week0004.java]: Have meaningful variable names instead of guest[*]

File: week0004/Week0004.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    public static ArrayList<Guest> read(String csvFilename) {
+        String line = "";
+        String cvsSplitBy = ",";
+        ArrayList<Guest> guests = new ArrayList<Guest>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(new File(csvFilename)));
+            for (int i = 0; (line = br.readLine()) != null; i++) {
+                if (i > 0) {
+                    // use comma as separator
```
Comment (vietlq) [inline week0004/Week0004.java]: Remove this comment or move it to line 63

**Comment by truongngodang** [comment]
know basic programming and study java since 1 month before :p I want breaking ice :v

**Comment by vietlq** [comment]
Are you student? Which year?

**Comment by truongngodang** [comment]
yes. Junior student :)

**Comment by vietlq** [comment]
Cool. Which university are you studying at?

**Comment by truongngodang** [comment]
I am studying multimedia at Posts and Telecommunications Institute of Technology VN 👍

**Comment by vietlq** [comment]
That's awesome! 😎

File: week0004/truongngodang/java/Week0004.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+        if (ReaderCVS.checkInputs(args)) {
+            ArrayList<Guest> guestsEvent1 = ReaderCVS.read(args[0]);
+            ArrayList<Guest> guestsEvent2 = ReaderCVS.read(args[1]);
+            Response response = new Response();
+            Checker checker = new Checker();
+            checker.check(guestsEvent1, guestsEvent2);
+            System.out.println(response.standardPrint(checker.getGuestErrors()));
+        } else {
+            System.out.println("File Not Allow!");
```
Comment (vietlq) [inline week0004/truongngodang/java/Week0004.java]: Which file is not allowed?

File: week0004/truongngodang/java/Week0004.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                    guests.add(new Guest(infos[0], infos[1], infos[2]));
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return guests;
+    }
+
+    public static boolean check(String fileName) {
```
Comment (vietlq) [inline week0004/truongngodang/java/Week0004.java]: I don't think we need to do this. I asked to check that the size of argv is 2 and the filenames are of valid files.

**Comment by vietlq** [comment]
You have an extra new line in your printing. Find it and remove it. Hint:

```
$ diff <(java Week0004 event1.input event2.input) guests.output
3d2
<
``

**Comment by truongngodang** [comment]
ok awesome 👍

**Comment by vietlq** [comment]
There's better way to do it ;) check the Main function ;)

**Pull request merged**